generator/booking_dataset.cpp

```diff
@@ -0,0 +1,242 @@
+#include "generator/booking_dataset.hpp"
+
+#include "indexer/search_delimiters.hpp"
+#include "indexer/search_string_utils.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "base/string_utils.hpp"
+
+#include "std/fstream.hpp"
+#include "std/iostream.hpp"
+#include "std/sstream.hpp"
+
+namespace generator
+{
+BookingDataset::Hotel::Hotel(string const & src)
+{
+  vector<string> rec(FieldsCount());
+  strings::SimpleTokenizer token(src, "\t");
+  for (size_t i = 0; token && i < rec.size(); ++i, ++token)
+    rec[i] = *token;
+
+  strings::to_uint(rec[Index(Fields::Id)], id);
+  strings::to_double(rec[Index(Fields::Latitude)], lat);
+  strings::to_double(rec[Index(Fields::Longtitude)], lon);
+  
+  name = rec[Index(Fields::Name)];
+  address = rec[Index(Fields::Address)];
+
+  strings::to_uint(rec[Index(Fields::Stars)], stars);
+  strings::to_uint(rec[Index(Fields::PriceCategory)], priceCategory);
+  strings::to_double(rec[Index(Fields::RatingBooking)], ratingBooking);
+  strings::to_double(rec[Index(Fields::RatingUsers)], ratingUser);
+
+  descUrl = rec[Index(Fields::DescUrl)];
+
+  strings::to_uint(rec[Index(Fields::Type)], type);
+}
+
+ostream & operator<<(ostream & s, BookingDataset::Hotel const & h)
+{
+  return s << "Name: " << h.name << " lat: " << h.lat << " lon: " << h.lon;
+}
+
+void BookingDataset::LoadHotels(string const & path)
+{
+  m_hotels.clear();
+
+  if (path.empty())
+    return;
+
+  ifstream src(path);
+  for (string line; getline(src, line);)
+    m_hotels.emplace_back(line);
+}
+
+BookingDataset::BookingDataset(string const & dataPath)
```

Comment:
The order of definitions should match the order of declarations. Ctor should be first.